src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,29 @@
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
+        Transaction transaction = session.beginTransaction();
+        try {
+            session.persist(country);
+            transaction.commit();
+        } catch (Exception e) {
```

Comment:
Remember to add a catch block for operations of all types on the DAO layer. In this case, catching a general `Exception` is appropriate, but it's also a good idea to log the exception to help with debugging.